mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1016,19 +1016,41 @@ struct GridwiseGemmV2RewritePattern
                << "bVectorLen: " << bVectorLen << "\n"
                << "vectorTiebreaker: " << vectorTiebreaker << "\n");
 
-    // Vectorization lengths evenly devide *CopyPerThread, this is safe.
-    int64_t aCopyKPerThread = aVectorDim == GemmDimension::K
-                                  ? aVectorLen
-                                  : aCopyPerThread / aVectorLen;
-    int64_t copyMPerThread = aVectorDim == GemmDimension::MorN
-                                 ? aVectorLen
-                                 : aCopyPerThread / aVectorLen;
-    int64_t bCopyKPerThread = bVectorDim == GemmDimension::K
-                                  ? bVectorLen
-                                  : bCopyPerThread / bVectorLen;
-    int64_t copyNPerThread = bVectorDim == GemmDimension::MorN
-                                 ? bVectorLen
-                                 : bCopyPerThread / bVectorLen;
+    int64_t aCopyKPerThread = 0;
+    int64_t copyMPerThread = 0;
+    if (aVectorDim == GemmDimension::K) {
+      aCopyKPerThread = std::min(aVectorLen, kPerBlock);
+      copyMPerThread = aCopyPerThread / aCopyKPerThread;
+    } else {
+      copyMPerThread = std::min(aVectorLen, mPerBlock);
+      aCopyKPerThread = aCopyPerThread / copyMPerThread;
+    }
+    if (aCopyKPerThread == 0 || copyMPerThread == 0) {
+      LLVM_DEBUG(llvm::dbgs() << "gemmA copy size too small.\n");
```

Comment:
These could maybe be `return op.emitOpError()` and possibly include the actual values via `<<`?

Reply:
Sounds good!